### PR TITLE
[GraphBolt] adapt GB's format to PyG

### PR DESCRIPTION
**Pull request by hutiechuan**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mfbalin** [comment]
It looks like this PR shows how to convert a GraphBolt dataset into a PyG dataset. Is there going to be more PRs in the future that convert GraphBolt sampled minibatches into PyG minibatches so that we can use the GraphBolt dataloader?

**Comment by hutiechuan** [comment]
Yes

**Comment by hutiechuan** [comment]
> It looks like this PR shows how to convert a GraphBolt dataset into a PyG dataset. Is there going to be more PRs in the future that convert GraphBolt sampled minibatches into PyG minibatches so that we can use the GraphBolt dataloader?

For sure

**Comment by Skeleton003** [comment]
Here's some simple steps to polish your code format using `lintrunner`.
0. `conda activate dgl` (or your env name)
1. `pip install lintrunner`
2. `lintrunner` or `lintrunner <file_name>`. The latter one shows all suggested code changes for a specific file while the former one are for all modified files (according to the output of `git status`.
3. `lintrunner -a`. This command will automatically apply the suggested changes to all modified files. Also, you can run `lintrunner -a <file_name>` whose effect is easy to guess.

File: examples/sampling/pyg/test.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    test_labels = test_set._items[1]
+    graph = dataset.graph
+    num_classes = dataset.tasks[0].metadata["num_classes"]
+    csc_indptr = graph.csc_indptr
+    rows = torch.repeat_interleave(torch.arange(csc_indptr.size(0) - 1, device=csc_indptr.device), csc_indptr[1:] - csc_indptr[:-1])
+    indices = graph.indices
+    cols = indices
+    edge_index =  torch.stack([rows, cols], dim=0)
+    pyg_data = Data(x=features_tensor, edge_index=edge_index, y=train_labels)  
+    num_nodes = csc_indptr.size(0) - 1
```
Comment (Skeleton003) [inline examples/sampling/pyg/test.py:48]: We already have api to get num_nodes https://github.com/dmlc/dgl/blob/master/python/dgl/graphbolt/impl/fused_csc_sampling_graph.py#L40

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 2]
```diff
@@ -599,7 +599,6 @@ class BuiltinDataset(OnDiskDataset):
     _all_datasets = _datasets + _large_datasets
 
     def __init__(self, name: str, root: str = "datasets") -> OnDiskDataset:
-        dataset_dir = os.path.join(root, name)
```
Comment (Skeleton003) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:602]: Why deleted?

**Comment by Skeleton003** [comment]
BTW it is better to checkout a new branch in your fork repo instead of using your master branch.

**Comment by hutiechuan** [comment]
> 3\. lintrunner -a

Yessir

**Comment by hutiechuan** [comment]
> BTW it is better to checkout a new branch in your fork repo instead of using your master branch.

will do

File: python/dgl/graphbolt/impl/ondisk_dataset.py (hunk 2 above)
Comment (hutiechuan) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:602]: Sorry, I accidentally deleted that line

File: examples/sampling/pyg/test.py (hunk 1 above)
Comment (hutiechuan) [reply examples/sampling/pyg/test.py:48]: got it, thanks!

**Comment by Rhett-Ying** [comment]
@hutiechuan pls close this PR if it's deprecated.

**Pull request closed without merging**